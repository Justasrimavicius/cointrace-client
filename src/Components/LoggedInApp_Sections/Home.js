import React, { useContext, useEffect, useState } from 'react'

import { Button, TableBody, TableCell, tableCellClasses, TableContainer, Typography } from "@mui/material";
import { Table, TableHead, TableRow, StyledTableCell } from "@mui/material";
import ArrowForwardIosIcon from '@mui/icons-material/ArrowForwardIos';
import ShoppingCartCheckoutIcon from '@mui/icons-material/ShoppingCartCheckout';
import { Avatar } from '@mui/material';
import { styled } from '@mui/material/styles';


import { UserContext } from '../AuthComponents/UserContext';

import { getDoc, doc } from 'firebase/firestore';
import { db } from '../Firebase-config';

import axios from 'axios';
import uniqid from 'uniqid';
import numeral from 'numeral';

function Home(props){
    const context = useContext(UserContext);
    const UID = context.UID;

    const [fiatAmount, setFiatAmount] = useState('00.00');
    const [coins, setCoins] = useState([]);
    let coinsNames = ['bitcoin', 'ethereum', 'polkadot', 'chainlink', 'cardano'];

    useEffect(()=>{ // updates fiat money in UI
        const userData = getDoc(doc(db, 'users', `${UID}`))
        .then((user)=>{
            if(user.data().fiat){
                setFiatAmount(user.data().fiat);
            } else {
                setFiatAmount('0.00');
            }
        })
    },[]);

    useEffect(()=>{
        coinsNames.forEach((coin)=>{
            axios.get(`https://api.coingecko.com/api/v3/coins/${coin}?localization=false&tickers=false&community_data=false&developer_data=false&sparkline=false`)
            .then((response)=>{
                setCoins(currentArr => [response.data, ...currentArr]);
            })
            .catch((error)=>{
            })
        })
    },[])

    const StyledTableCell = styled(TableCell)(({ theme }) => ({
        [`&.${tableCellClasses.head}`]: {
          backgroundColor: theme.palette.primary.light,
          color: theme.palette.common.white,
        },
        [`&.${tableCellClasses.body}`]: {
          fontSize: 14,
        },
      }));
    return(
        <React.Fragment>
            <div className='home'>
                <div className="middle">
                <div className="middle-balance">
                    <Typography variant="h6">Your balance</Typography>
                    <Typography variant="h3" className="middle-balance-number" >{fiatAmount}$<ArrowForwardIosIcon className='home-arrow' style={{marginBottom:'4px'}} onClick={()=>{props.loadFromSidePanel.loadSidePanelTab('portfolio')}}/></Typography>
                    
                </div>
                <div className="middle-coins">
                    <div className="middle-coins-top">
                    <Typography variant="h6" style={{marginLeft: '5px'}}>Popular coins</Typography>
                    <ShoppingCartCheckoutIcon style={{margin: '5px'}} className='middle-coins-top-icon' onClick={()=>{props.loadFromSidePanel.loadSidePanelTab('trade')}}/>
                    </div>
                    <div className="middle-coins-bottom">
                        <TableContainer  style={{overflow:'auto', height:'100%'}}>
                            <Table height='100%' data-testid='home-table'>
                                <TableHead>
                                    <TableRow>
                                        <StyledTableCell>Name</StyledTableCell>
                                        <StyledTableCell align='right'>Current price $</StyledTableCell>
                                        <StyledTableCell align='right'>Market cap %</StyledTableCell>
                                        <StyledTableCell align='right'>Market cap change 24h %</StyledTableCell>
                                        <StyledTableCell>Purchase</StyledTableCell>
                                    </TableRow>
                                </TableHead>
                                <TableBody style={{overflow:'hidden'}}>
                                {coins!=[] ? 
                                    coins.map(coin=>{
                                        return (
                                        <TableRow key={uniqid()}>
                                            <StyledTableCell align="right" style={{textAlign:'center', minHeight:'40px', maxHeight:'40px',borderBottom:'1px solid gray'}} ><Avatar src={`${coin.image.small}`} style={{width:'25px', height:'25px',transform:'translate(-50%)', marginLeft: '50%'}} />{coin.name}</StyledTableCell>
                                            <StyledTableCell align="right">{coin.market_data.current_price.usd.toFixed(2)}$</StyledTableCell>
                                            <StyledTableCell align="right">{numeral(coin.market_data.market_cap.usd).format('0.00a')}$</StyledTableCell>
                                            {coin.market_data.market_cap_change_percentage_24h > 0 ? <StyledTableCell align="right" style={{color:'green'}}>{coin.market_data.market_cap_change_percentage_24h.toFixed(2)}%</StyledTableCell>
                                            : <StyledTableCell align="right" style={{color:'red'}}>{coin.market_data.market_cap_change_percentage_24h.toFixed(2)}%</StyledTableCell>}
                                            <StyledTableCell style={{borderBottom:'1px solid gray'}}><Button variant="outlined" style={{height:'30px'}} onClick={()=>{props.loadFromSidePanel.loadSidePanelTab('trade')}}>Buy</Button></StyledTableCell>
                                        </TableRow>)
                                    }) : null}
                                </TableBody>
                            </Table>
                        </TableContainer>
                    </div>
                </div>
                </div>
                <div className="right-side">
                <div className="right-side-upper">
                    <Button variant="text" align="center" fullWidth onClick={()=>{window.open('https://cryptoliteracy.org/quiz/')}}>Check your crypto knowledge!</Button>
                </div>
                <div className="right-side-lower">
                    <div className="nft">
                        <Typography variant="subtitle1" style={{alignSelf:'center', margin:'10px'}}>What are NFTs and where did they come from?</Typography>
                        <img src={require('../../Images/nft.png')} width='140px' style={{alignSelf:'center'}} onClick={()=>{window.open('https://www.investopedia.com/non-fungible-tokens-nft-5115211')}}></img>
                    </div>
                    <div className="staking">
                        <img src={require('../../Images/staking.png')} width='140px' style={{alignSelf:'center'}} onClick={()=>{window.open('https://www.coindesk.com/learn/crypto-staking-101-what-is-staking/')}}></img>
                        <Typography variant="subtitle1" style={{alignSelf:'center', margin:'10px'}}>What is staking and why is it so amazing?</Typography>
                    </div>
                    <div className="blockchain">
                        <Typography variant="subtitle1" style={{alignSelf:'center', margin:'10px'}}>How the allmighty blockchain works</Typography>
                        <img src={require('../../Images/blockchain.png')} width='140px' style={{alignSelf:'center'}} onClick={()=>{window.open('https://www.investopedia.com/terms/b/blockchain.asp')}}></img>
                    </div>
                </div>
                </div>
                <div className="footer">
                <span>@2022 <a href="https://github.com/Justasrimavicius/" style={{textDecoration:'none'}}>https://github.com/Justasrimavicius/</a></span>
                <Button variant="text" color="secondary" onClick={()=>{window.open('https://www.youtube.com/watch?v=dQw4w9WgXcQ')}}>Click me for a surprise</Button>
                </div>
            </div>
        </React.Fragment>
    )
}

export default Home;