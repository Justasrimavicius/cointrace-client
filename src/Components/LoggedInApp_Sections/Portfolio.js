import { Dialog, DialogTitle, Typography } from "@mui/material";

import { db } from "../Firebase-config";
import { setDoc, doc, getDoc, getDocs, updateDoc, collection } from "firebase/firestore";

import { useState, useEffect } from "react";
import * as React from 'react';
import { styled } from '@mui/material/styles';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell, { tableCellClasses } from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import { Button } from "@mui/material";
import { TextField }from "@mui/material";

import uniqid from 'uniqid';
import axios from "axios";

import { useContext } from "react";
import { UserContext } from "../AuthComponents/UserContext";
import { GetApp } from "@mui/icons-material";

function Portfolio(props){
    const [topupBtn, disableTopupBtn] = useState('text');
    const [alert, setAlert] = useState(null);

    const [balanceTopup, setBalanceTopup] = useState(null);
    const [fiatAmount, setFiatAmount] = useState('0.00');
    const [mockFiatAmount, setMockFiatAmount] = useState('0');
    const [coins, setCoins] = useState([]);
    const [coinsDisplay, setCoinsDisplay] = useState(null);

    const context = useContext(UserContext);
    const UID = context.UID;

    const StyledTableCell = styled(TableCell)(({ theme }) => ({
      [`&.${tableCellClasses.head}`]: {
        backgroundColor: theme.palette.primary.light,
        color: theme.palette.common.white,
      },
      [`&.${tableCellClasses.body}`]: {
        fontSize: 14,
      },
    }));

    useEffect(()=>{
        if(topupBtn=='disabled'){
            setDoc(doc(db, "users", `${UID}`), {topupBtn: `${topupBtn}`}, {merge: true});
        }
    },[topupBtn])
    useEffect(()=>{
        const userData = getDoc(doc(db, 'users', `${UID}`))
        .then((user)=>{
            if(user.data().fiat){
                setFiatAmount(parseFloat(user.data().fiat).toFixed(2));
            } else {
                setFiatAmount('0.00');
            }

            if(user.data().topupBtn=='disabled'){
                disableTopupBtn('disabled');
            }
        })

    },[]);

    useEffect(()=>{

        if(fiatAmount!='0.00'){
            updateDoc(doc(db, "users", `${UID}`), {fiat: `${fiatAmount}`});
        }
    },[fiatAmount])

    useEffect(()=>{
        if(alert!=null){
            setTimeout(() => {
                setAlert(null);
            }, 4000);
        }
    },[alert])

    function HandleTopup(){
        // check for too large topup amounts(100mil max)
        if(!mockFiatAmount.includes('.') && mockFiatAmount.length>9){
            setAlert('Topup sum is too big');
            return;
        } else if(mockFiatAmount.split('.')[0].length>9){
            setAlert('Topup sum is too big');
            return;
        }

        // add 0s to the end for better UI view
        if(!mockFiatAmount.includes('.')){

            setFiatAmount(mockFiatAmount+'.00');

        } 
        else if(mockFiatAmount.split('.')[1].length==1){
            setFiatAmount(mockFiatAmount+'0');
        }
        else if(mockFiatAmount.split('.')[1].length>2){
            setFiatAmount(mockFiatAmount.split('.')[0]+'.'+mockFiatAmount.split('.')[1].substring(0,2))
        } else {
            setFiatAmount(mockFiatAmount);
        }
        disableTopupBtn('disabled');
        setBalanceTopup(false);
}

    useEffect(()=>{
        getDocs(collection(db, `users/${UID}/crypto`))
        .then(result=>{
            async function getCoin(){
            for(let document of result.docs){
                const name = (document.data().name).toLowerCase();
                const amount = document.data().amount;
                let price;
                let avatarSrc;
                let priceChange24h;
                await axios.get(`https://api.coingecko.com/api/v3/coins/${name}?localization=false&tickers=false&community_data=false&developer_data=false&sparkline=false`)
                .then(result=>{
                    price=result.data.market_data.current_price.usd;
                    avatarSrc=result.data.image.small;
                    priceChange24h=result.data.market_data.price_change_percentage_24h;
                    let coin = {
                        name: `${name}`,
                        amount: `${amount}`,
                        price: `${price}`,
                        priceChange24h: `${priceChange24h}`,
                        avatarSrc: `${avatarSrc}`,
                    }
                    setCoins(currentArr => [coin, ...currentArr]);
                })
            }
                }
                getCoin();
        })

    },[])

    useEffect(()=>{
        console.log(coins);
        setCoinsDisplay(true);
    },[coins])

    return(
        <div className="portfolio">
            <div className="portfolio-left">
                <div className="portfolio-left-upper">
                    <Typography align="center" paddingTop='10px'>Fiat balance:</Typography>
                    <Typography variant="h3" className="portfolio-left-balance" align="center">{fiatAmount}$</Typography>
                </div>
                <div className="portfolio-left-coins">
                    <TableContainer style={{height:'75vh'}}>
                        <Table stickyHeader aria-label="customized table">
                            <TableHead>
                                <TableRow>
                                    <StyledTableCell align="left">Name</StyledTableCell>
                                    <StyledTableCell align="right">Balance</StyledTableCell>
                                    <StyledTableCell align="right">Price of 1</StyledTableCell>
                                    <StyledTableCell align="right">24h price change %</StyledTableCell>
                                </TableRow>
                            </TableHead>
                            <TableBody>
                                {coinsDisplay ? coins.map((coin)=>{
                                    return(<TableRow key={uniqid()}>
                                       <StyledTableCell align="left">{coin.name}</StyledTableCell>
                                       <StyledTableCell align="right">{coin.amount}</StyledTableCell>
                                       <StyledTableCell align="right">{coin.price}$</StyledTableCell>
                                       <StyledTableCell align="right">{parseFloat(coin.priceChange24h).toFixed(2)}%</StyledTableCell>
                                    </TableRow>)
                                }) : null }
                            </TableBody>
                        </Table>
                        {coins.length==0 ? <Button fullWidth onClick={()=>{props.loadFromSidePanel.loadSidePanelTab('trade')}}>No coins in your portfolio. Add some!</Button> : null}

                    </TableContainer>
                </div>
            </div>
            <div className="portfolio-right">
                <Button fullWidth style={{height:'161px', borderRadius:'0px', borderBottom:'1px solid black'}} variant={topupBtn} className='topup-button' onClick={()=>{setBalanceTopup(true)}} >Top up your balance</Button>
            </div>
            {balanceTopup ? 
                <Dialog open={true}>
                    <DialogTitle align="center">Type in an amount</DialogTitle>
                        <DialogTitle align="center">You can only topup your balance once - we recommend you insert a high amount(maximum 100 millions)</DialogTitle>
                    <TextField type='number' variant="filled" className="fiat-amount" placeholder="100000.00" onChange={(e)=>{setMockFiatAmount(e.target.value)}}></TextField>
                    <div style={{display:'flex',flexDirection:'row', justifyContent:'space-around'}}>
                        <Button variant="outlined" fullWidth onClick={()=>{HandleTopup()}}>Continue</Button>
                        <Button variant="outlined" onClick={()=>{setBalanceTopup(false)}}>X</Button>
                    </div>
                </Dialog>
             : null}
             {alert ? <Dialog open={true}><DialogTitle>{alert}</DialogTitle></Dialog> : null}
        </div>
    )
}

export default Portfolio;