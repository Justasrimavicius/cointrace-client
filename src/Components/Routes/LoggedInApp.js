import React, { useEffect, useState } from "react";
import Header from "../Header";
import Home from "../LoggedInApp_Sections/Home";
import Portfolio from '../LoggedInApp_Sections/Portfolio';
import Trade from '../LoggedInApp_Sections/Trade';

import { Button, List, ListItem, ListItemText, Divider, ListItemButton, ListItemIcon, Typography } from "@mui/material";
import ShowChartIcon from '@mui/icons-material/ShowChart';
import HomeIcon from '@mui/icons-material/Home';
import AllInboxIcon from '@mui/icons-material/AllInbox';
import { UserContext } from "../AuthComponents/UserContext";
import { useNavigate } from "react-router-dom";

function LoggedInApp(){

    const [sidePanelTab, loadSidePanelTab] = useState('home');
    const navigate = useNavigate();
    let UID = sessionStorage.getItem('user');




    return(
        <React.Fragment>
            <UserContext.Provider value={{UID}}>
            <Header props={{loggedIn:true}}/>

            <div className="logged-in-app" data-testid='LoggedInApp'>
                <div className="left-side">
                    <List style={{paddingTop:'0'}} className="left-side-list">

                        <ListItemButton onClick={()=>{loadSidePanelTab('home')}}>
                            <ListItemIcon><HomeIcon/></ListItemIcon>
                            <ListItem>
                                <ListItemText>Home</ListItemText>
                            </ListItem>
                        </ListItemButton>

                        <ListItemButton onClick={()=>{loadSidePanelTab('portfolio')}}>
                            <ListItemIcon><AllInboxIcon/></ListItemIcon>
                            <ListItem>
                                <ListItemText>Portfolio</ListItemText>
                            </ListItem>
                        </ListItemButton>

                        <ListItemButton onClick={()=>{loadSidePanelTab('trade')}}>
                        <ListItemIcon><ShowChartIcon/></ListItemIcon>
                            <ListItem>
                                <ListItemText>Trade</ListItemText>
                            </ListItem>
                        </ListItemButton>

                        <Divider />
                    </List>
                    <Button variant="contained" color="secondary" style={{borderRadius:'5px 5px 0px 5px'}} onClick={()=>{navigate('/')}}>Log out</Button>
                </div>
                {sidePanelTab=='home' ? <Home loadFromSidePanel={{sidePanelTab, loadSidePanelTab}}/> : null}
                {sidePanelTab=='portfolio' ? <Portfolio loadFromSidePanel={{sidePanelTab, loadSidePanelTab}}/> : null}
                {sidePanelTab=='trade' ? <Trade /> : null}


            </div>
            </UserContext.Provider>
        </React.Fragment>

    )
}

export default LoggedInApp;