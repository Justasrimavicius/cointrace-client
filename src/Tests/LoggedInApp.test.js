import { render, screen, cleanup, afterEach, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from "react-router-dom";
import { toBeInTheDocument } from '@testing-library/jest-dom/dist/matchers';

import renderer from 'react-test-renderer';

import LoggedInApp from '../Components/Routes/LoggedInApp'
import Home from '../Components/LoggedInApp_Sections/Home';
import Trade from '../Components/LoggedInApp_Sections/Trade';
import Portfolio from '../Components/LoggedInApp_Sections/Portfolio';


test('renders header', () => {

    render(<Router><LoggedInApp/></Router>)

    const loggedInApp = screen.getByTestId('LoggedInApp');
    expect(loggedInApp).toBeInTheDocument();
});

describe('LoggedInApp sections',()=>{
    test('home section',()=>{
        const tree = renderer.create(<Home />).toJSON();
        expect(tree).toMatchSnapshot();
    });
    test('table of crypto coins loads in home',async ()=>{
        render(<Home />)
        const table = screen.getByTestId('home-table');
        expect(table).toBeInTheDocument();
    });
    test('portfolio section',()=>{
        const tree = renderer.create(<Portfolio />).toJSON();
        expect(tree).toMatchSnapshot();
    })
    test('trade section',()=>{
        const tree = renderer.create(<Trade />).toJSON();
        expect(tree).toMatchSnapshot();
    })
});
