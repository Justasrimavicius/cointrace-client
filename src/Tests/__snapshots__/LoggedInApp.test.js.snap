// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoggedInApp sections home section 1`] = `
<div
  className="home"
>
  <div
    className="middle"
  >
    <div
      className="middle-balance"
    >
      <h6
        className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        Your balance
      </h6>
      <h3
        className="MuiTypography-root MuiTypography-h3 middle-balance-number css-gepadz-MuiTypography-root"
      >
        00.00
        $
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium home-arrow css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowForwardIosIcon"
          focusable="false"
          onClick={[Function]}
          style={
            Object {
              "marginBottom": "4px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M6.23 20.23 8 22l10-10L8 2 6.23 3.77 14.46 12z"
          />
        </svg>
      </h3>
    </div>
    <div
      className="middle-coins"
    >
      <div
        className="middle-coins-top"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          style={
            Object {
              "marginLeft": "5px",
            }
          }
        >
          Popular coins
        </h6>
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium middle-coins-top-icon css-i4bv87-MuiSvgIcon-root"
          data-testid="ShoppingCartCheckoutIcon"
          focusable="false"
          onClick={[Function]}
          style={
            Object {
              "margin": "5px",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2zm10 0c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2zm-8.9-5h7.45c.75 0 1.41-.41 1.75-1.03L21 4.96 19.25 4l-3.7 7H8.53L4.27 2H1v2h2l3.6 7.59-1.35 2.44C4.52 15.37 5.48 17 7 17h12v-2H7l1.1-2zM12 2l4 4-4 4-1.41-1.41L12.17 7H8V5h4.17l-1.59-1.59L12 2z"
          />
        </svg>
      </div>
      <div
        className="middle-coins-bottom"
      >
        <div
          className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
          style={
            Object {
              "height": "100%",
              "overflow": "auto",
            }
          }
        >
          <table
            className="MuiTable-root css-rqglhn-MuiTable-root"
            data-testid="home-table"
            height="100%"
            role={null}
          >
            <thead
              className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
              role={null}
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
                role={null}
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-twzqbi-MuiTableCell-root"
                  scope="col"
                >
                  Name
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
                  scope="col"
                >
                  Current price $
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
                  scope="col"
                >
                  Market cap %
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
                  scope="col"
                >
                  Market cap change 24h %
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-twzqbi-MuiTableCell-root"
                  scope="col"
                >
                  Purchase
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
              role={null}
              style={
                Object {
                  "overflow": "hidden",
                }
              }
            />
          </table>
        </div>
      </div>
    </div>
  </div>
  <div
    className="right-side"
  >
    <div
      className="right-side-upper"
    >
      <button
        align="center"
        className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth css-10vjzfb-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        Check your crypto knowledge!
      </button>
    </div>
    <div
      className="right-side-lower"
    >
      <div
        className="nft"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          style={
            Object {
              "alignSelf": "center",
              "margin": "10px",
            }
          }
        >
          What are NFTs and where did they come from?
        </h6>
        <img
          onClick={[Function]}
          src="nft.png"
          style={
            Object {
              "alignSelf": "center",
            }
          }
          width="140px"
        />
      </div>
      <div
        className="staking"
      >
        <img
          onClick={[Function]}
          src="staking.png"
          style={
            Object {
              "alignSelf": "center",
            }
          }
          width="140px"
        />
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          style={
            Object {
              "alignSelf": "center",
              "margin": "10px",
            }
          }
        >
          What is staking and why is it so amazing?
        </h6>
      </div>
      <div
        className="blockchain"
      >
        <h6
          className="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
          style={
            Object {
              "alignSelf": "center",
              "margin": "10px",
            }
          }
        >
          How the allmighty blockchain works
        </h6>
        <img
          onClick={[Function]}
          src="blockchain.png"
          style={
            Object {
              "alignSelf": "center",
            }
          }
          width="140px"
        />
      </div>
    </div>
  </div>
  <div
    className="footer"
  >
    <span>
      @2022 
      <a
        href="https://github.com/Justasrimavicius/"
        style={
          Object {
            "textDecoration": "none",
          }
        }
      >
        https://github.com/Justasrimavicius/
      </a>
    </span>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium css-1d3f8j8-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Click me for a surprise
    </button>
  </div>
</div>
`;

exports[`LoggedInApp sections portfolio section 1`] = `
<div
  className="portfolio"
>
  <div
    className="portfolio-left"
  >
    <div
      className="portfolio-left-upper"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-14h5f3b-MuiTypography-root"
      >
        Fiat balance:
      </p>
      <h3
        className="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter portfolio-left-balance css-47suzj-MuiTypography-root"
      >
        0.00
        $
      </h3>
    </div>
    <div
      className="portfolio-left-coins"
    >
      <div
        className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
        style={
          Object {
            "height": "75vh",
          }
        }
      >
        <table
          aria-label="customized table"
          className="MuiTable-root MuiTable-stickyHeader css-xn82ks-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1x9ecgr-MuiTableCell-root"
                scope="col"
              >
                Name
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-1cqtko9-MuiTableCell-root"
                scope="col"
              >
                Balance
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-1cqtko9-MuiTableCell-root"
                scope="col"
              >
                Price of 1
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-alignRight MuiTableCell-sizeMedium css-1cqtko9-MuiTableCell-root"
                scope="col"
              >
                24h price change %
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
        </table>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth css-10vjzfb-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          No coins in your portfolio. Add some!
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="portfolio-right"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth topup-button css-10vjzfb-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "borderBottom": "1px solid black",
          "borderRadius": "0px",
          "height": "161px",
        }
      }
      tabIndex={0}
      type="button"
    >
      Top up your balance
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`LoggedInApp sections trade section 1`] = `
<div
  className="trade"
  style={
    Object {
      "width": "100%",
    }
  }
>
  <h6
    className="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
  >
    Your balance
  </h6>
  <h3
    className="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-47suzj-MuiTypography-root"
  >
    0.00
    $
  </h3>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root trade-search css-wb57ya-MuiFormControl-root-MuiTextField-root"
    style={
      Object {
        "height": "50px",
      }
    }
  >
    <div
      className="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-gjbq6i-MuiInputBase-root-MuiFilledInput-root"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        autoFocus={false}
        className="MuiInputBase-input MuiFilledInput-input css-10botns-MuiInputBase-input-MuiFilledInput-input"
        disabled={false}
        id=":r0:"
        onAnimationStart={[Function]}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder="Bitcoin, ethereum, solana, etc."
        required={false}
        type="text"
      />
    </div>
  </div>
  <div
    className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
    style={
      Object {
        "marginTop": "5px",
      }
    }
  >
    <table
      aria-label="customized table"
      className="MuiTable-root css-rqglhn-MuiTable-root"
      role={null}
      style={
        Object {
          "overflow": "auto",
        }
      }
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-twzqbi-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
            scope="col"
          >
            Price
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
            scope="col"
          >
            Market cap
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
            scope="col"
          >
            24h change
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-sizeMedium css-1ef1k85-MuiTableCell-root"
            scope="col"
          >
            Purchase
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      />
    </table>
  </div>
</div>
`;
